### ISSUE TITLE
Work with Hardware Token

### ISSUE THREAD

**Issue by igloo15** [Daniel15/SecureSign#9]
Could this work with a hardware token. For instance if I have a machine setup with the token connected to it and then setup this software on that machine could I get this to work?

**Comment by Daniel15** [comment]
Sorry, I don't have any experience with hardware tokens, so this likely won't work with them as-is. I mainly designed SecureSign to run on VPS providers where I don't have physical access to the hardware. By hardware token, do you mean a token that contains an Authenticode/GPG certificate?

**Comment by igloo15** [comment]
Yes typically we have a laptop or desktop computer that has a hardware token attached to it via usb. When sign tool is launched a popup occurs asking us to input a passcode and then the cert is loaded from the token to sign the code. I was hoping this could potentially be augmented so that people could use this software to do the signing over the network. I planned to install this software on the laptop with the token attached. If entered the passcode it should be able to access the cert as long as the application doesn't close or the computer restarts.

**Comment by hrumhurum** [comment]
[adhoc-file-signer](https://github.com/gapotchenko/adhoc-file-signer) project supports hardware tokens in unattended mode. Admittedly, HSMs can be challenging to set up, as noted in the [Getting Started Guide](https://github.com/gapotchenko/adhoc-file-signer/tree/main/docs/guides/getting-started/windows). Hopefully, this will help others save weeks of effort.

Disclaimer: I am the project author.
